### PR TITLE
Implement unread update indicator

### PR DESCRIPTION
**Pull request by tfussell**

### CODE REVIEW

File: src/module/interface/user/index.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export interface IUser {
   name: string;
   title?: string;
   job?: Job;
+  lastUpdate?: Record<string, string>;
```
Comment (tfussell) [inline src/module/interface/user/index.ts:14]: The new metadata property on user objects indicating the last update viewed for each timeline.

File: src/module/interface/timeline/index.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ export interface ITimeline {
   ventureId: string;
   userId?: string;
   membersWrite: boolean;
+  lastUpdate?: string;
```
Comment (tfussell) [inline src/module/interface/timeline/index.ts:10]: New metadata property returned by the `apiserver` (in https://github.com/venturemark/apiserver/pull/235) which indicates the ID of the most recent update in the timeline.

File: src/module/hook/user/index.ts [hunk 3]
```diff
@@ -144,14 +145,32 @@ export function useUpdateUser() {
...
-  return useMutation<IUser[], any, IUpdateUser>(
+  return useMutation<UpdateStatus[], any, IUpdateUser>(
     (userUpdate: IUpdateUser) => {
       return api.API.User.Update(userUpdate);
     },
     {
       onSuccess: (data, userUpdate) => {
-        // Invalidate and refetch
-        queryClient.invalidateQueries("users");
+        if (
```
Comment (tfussell) [inline src/module/hook/user/index.ts:154]: Optimization for updating update viewed status without invalidating everything (showing flash of unloaded state).

File: src/module/apikeys/index.ts [hunk 4]
```diff
@@ -21,3 +21,5 @@ export const ResourceKind = "resource.venturemark.co/kind";
 export const InviteCode = "invite.venturemark.co/code";
 
 export const PermissionModel = "permission.venturemark.co/model";
+
+export const TimelineLastUpdate = "timeline.venturemark.co/last-update";
```
Comment (tfussell) [inline src/module/apikeys/index.ts:25]: Corresponds to https://github.com/venturemark/apicommon/pull/41

File: src/module/api/invite/delete/index.ts [hunk 5]
```diff
@@ -30,14 +33,12 @@ export async function Delete(IDeleteInvite: IDeleteInvite): Promise<IInvite[]> {
         reject(err);
         return;
       } else {
-        const invitesPb = res.getObjList();
-
-        const status = invitesPb.map((invitePb) => {
-          const metaPb = invitePb.getMetadataMap();
-          const id = metaPb.get(key.InviteStatus);
-          return id as unknown as IInvite; // TODO: check
```
Comment (tfussell) [inline src/module/api/invite/delete/index.ts:38]: `key.*Status` has always been one of `updated`, `created`, or `deleted`. We ignored the return value so this wasn't noticed before. I changed this in every mutating API call.

File: src/component/feedupdate/index.tsx [hunk 6]
```diff
@@ -61,47 +76,94 @@ function FeedUpdate(props: FeedUpdateProps) {
...
+  }, [
+    timelineId,
+    timelineUpdates,
+    timelineUpdatesSuccess,
+    ventureId,
+    ventureUpdates,
+    ventureUpdatesSuccess,
+  ]);
+
+  useEffect(() => {
```
Comment (tfussell) [inline src/component/feedupdate/index.tsx:115]: Here's the hook that updates the update viewed status for the user. We assume once the update has been rendered here then the user has seen it.

File: src/component/app/VentureRoutes.tsx [hunk 7]
```diff
@@ -83,140 +36,147 @@ export function VentureRoutes(props: VentureRoutesProps) {
...
         dataLayerName: "UserDataLayer",
       };
       TagManager.dataLayer(tagManagerArgs);
     }
   }, [userId]);
 
-  if (venturesLoaded && !venturesWithRoles.length) {
-    return <Navigate replace to={`/begin`} />;
-  }
+  const { data: venturesData = [], status: venturesStatus } = useVenturesByUser(
```
Comment (tfussell) [inline src/component/app/VentureRoutes.tsx:86]: Cleaning up the venture context to simplify data fetching.

**Review by marcusellison — APPROVED** [review]

**Pull request merged**